enterprise_access/apps/subsidy_request/models.py

```diff
@@ -306,3 +307,86 @@ def update_course_info_for_subsidy_request(sender, **kwargs):  # pylint: disable
         subsidy_type,
         str(subsidy_request.uuid),
     )
+
+
+class LearnerCreditRequest(SubsidyRequest):
+    """
+    Stores information related to a learner credit request.
+
+    .. no_pii: This model has no PII
+    """
+
+    subsidy_access_policy = models.ForeignKey(
+        SubsidyAccessPolicy,
+        related_name="learner_credit_requests",
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        help_text="The SubsidyAccessPolicy that this learner credit request is associated with.",
+    )
+
+    assignment = models.OneToOneField(
+        'content_assignments.LearnerContentAssignment',  # pylint: disable=all
+        related_name="credit_request",
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        help_text="The content assignment associated with this request."
+    )
+
+    history = HistoricalRecords()
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=['user', 'enterprise_customer_uuid', 'course_id'],
+                name='unique_learner_course_request',
+                condition=models.Q(state__in=[SubsidyRequestStates.REQUESTED, SubsidyRequestStates.PENDING]),
+            )
+        ]
+
+    def __str__(self):
+        """
+        Return human-readable string representation.
+        """
+        if self.course_id:
+            return f'<LearnerCreditRequest for user {self.user} and course {self.course_id}>'
+        return f'<LearnerCreditRequest for user {self.user}>'
+
+    def approve(self, reviewer):
+        self.reviewer = reviewer
+        self.state = SubsidyRequestStates.PENDING
+        self.reviewed_at = localized_utcnow()
+        self.save()
+
+    def decline(self, reviewer, reason=None):
+        self.reviewer = reviewer
+        self.state = SubsidyRequestStates.DECLINED
+        self.decline_reason = reason
+        self.reviewed_at = localized_utcnow()
+        self.save()
+
+
+class LearnerCreditRequestConfiguration(TimeStampedModel):
```

Comment:
Would it make sense to include a cached_property for `enterprise_customer_uuid`?